kong/clustering/compat/checkers.lua

```diff
@@ -23,6 +24,47 @@ end
 
 
 local compatible_checkers = {
+  { 3007000000, --[[ 3.7.0.0 ]]
+    function(config_table, dp_version, log_suffix)
+      local has_update
+
+      for _, route in ipairs(config_table.routes or {}) do
+        local snis = route.snis
+        if type(snis) ~= "table" or #snis == 0 then
+          goto continue
+        end
+
+        local idx = 0
+        local new_snis = {}
+        local local_has_update
+        for _, sni in ipairs(snis) do
+          -- remove the sni that contains wildcard
+          if string_find(sni, "*", nil, true) then
+            has_update = true
+            local_has_update = true
+
+          else
+            idx = idx + 1
+            new_snis[idx] = sni
+          end
```

Comment:
This may create security vulnerabilities because expected routes no longer match and just fail silently. We should consider not sending the config and output an error instead.

Reply:
@mheap @veenaraja Please take a look^

Reply:
Michael is also leaning towards the option 1. I will remove the compatibility check code.

Reply:
Where is the action "refuse the whole config and directly error out"?

Reply:
I think DPs will [refuse the config](https://github.com/Kong/kong/blob/master/kong/clustering/data_plane.lua#L254) on [validation failure](https://github.com/Kong/kong/blob/master/kong/db/schema/others/declarative_config.lua#L739) by default and [send the error info](https://github.com/Kong/kong/blob/master/kong/clustering/data_plane.lua#L288) to the CP.